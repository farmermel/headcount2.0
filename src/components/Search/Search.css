.search-wrap {
  width: 56%;
}

.search {
  padding: 2px 0 2px 10px;
  height: 26px;
  font-size: 1em;
  width: 100%;
}