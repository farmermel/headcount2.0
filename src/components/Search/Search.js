import React from 'react';
import PropTypes from 'prop-types';
import './Search.css';

const Search = ({ searchDistricts }) => {
  return (
    <div className="search-wrap">
      <input
        type="text"
        placeholder="Search Districts"
        className="search"
        onChange={e => searchDistricts(e.target.value)}
      />
    </div>
  );
};

Search.propTypes = {
  searchDistricts: PropTypes.func.isRequired
};

export default Search;
