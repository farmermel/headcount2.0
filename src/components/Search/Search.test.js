import React from 'react';
import ReactDom from 'react-dom';
import Search from './Search';
import { shallow } from 'enzyme';

describe('Search', () => {
  const mockSearchDistricts = jest.fn();

  it('should match the snapshot', () => {
    const wrapper = shallow(<Search searchDistricts={mockSearchDistricts} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call searchDisctricts on change passing in the input values', () => {
    const wrapper = shallow(<Search searchDistricts={mockSearchDistricts} />);
    const mockEvent = { target: { value: 'Chicken' } };
    wrapper.find('.search').simulate('change', mockEvent);
    expect(mockSearchDistricts).toHaveBeenCalledWith('Chicken');
  });
});
