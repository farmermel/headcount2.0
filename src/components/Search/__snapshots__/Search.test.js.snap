// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    searchDistricts={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        className="search"
        onChange={[Function]}
        placeholder="Search Districts"
        type="text"
/>,
      "className": "search-wrap",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "search",
        "onChange": [Function],
        "placeholder": "Search Districts",
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          className="search"
          onChange={[Function]}
          placeholder="Search Districts"
          type="text"
/>,
        "className": "search-wrap",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "search",
          "onChange": [Function],
          "placeholder": "Search Districts",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
